yktracec/src/jitmodbuilder.cc

```diff
@@ -1109,14 +1115,27 @@ class JITModBuilder {
           continue;
         } else if (LastInst && isa<ReturnInst>(LastInst)) {
           LastInst = nullptr;
-          assert(CallStack.back()->getParent() == BB);
-          LastBB = BB;
+          if (!IsSWTrace) {
+            assert(CallStack.back()->getParent() == BB);
+          }
+          LastBB = CallStack.back()->getParent();
           CallStack.pop_back();
           if (CallStack.size() == OutlineBase) {
             Outlining = false;
             OutlineBase = 0;
           }
-          continue;
+          if (!IsSWTrace)
+            continue;
+        }
+        if (IsSWTrace) {
+          // In the software tracer external calls are invisible. So when an
```

Comment:
I'm less sure than with https://github.com/ykjit/yk/pull/939/files#r1481895717, but I think this part of the code might also suffer from the same problem (i.e. mappable -> unmappable -> mappable code).